contracts/Reserve/AmmReserve/AmmReserve.cpp

```diff
@@ -64,12 +64,25 @@ ACTION AmmReserve::setparams(double r,
                              name   fee_wallet) {
     get_state_assert_admin();
 
-    eosio_assert(profit_percent < 100.0, "illegal profit_percent");
-    eosio_assert(min_sell_rate < max_sell_rate, "min_sell_rate not smaller than max_sell_rate");
+    eosio_assert(r > 0 && r < 100.0, "illegal r");
```

Comment:
I was just thinking about it today that for a promotion coupon we need r = 0. also why force r < 100?

Reply:
will do only r>=0